pkg/nfd-master/nfd-master.go

```diff
@@ -605,19 +605,74 @@ func (m *nfdMaster) nfdAPIUpdateOneNode(nodeName string) error {
 	return nil
 }
 
-func (m *nfdMaster) refreshNodeFeatures(cli *kubernetes.Clientset, nodeName string, annotations, labels map[string]string, features *nfdv1alpha1.Features) error {
+// validateExtendedResources validates extended resources and returns a map
+// of valid extended resources.
+func (m *nfdMaster) validateExtendedResources(features *nfdv1alpha1.Features, resourceLabels map[string]string) map[string]string {
+	extendedResources := make(map[string]string)
+	for extendedResource, capacity := range resourceLabels {
+		// Add possibly missing default ns
+		extendedResource = addNs(extendedResource, nfdv1alpha1.ExtendedResourceNs)
+		// Check if given NS is allowed
+		ns, _ := splitNs(extendedResource)
+		if isNamespaceDenied(ns, m.deniedNs.wildcard, m.deniedNs.normal) {
+			if _, ok := m.args.ExtraLabelNs[ns]; !ok {
+				klog.Errorf("Namespace %q is not allowed. Ignoring Extended Resource  %q\n", ns, extendedResource)
+				continue
+			}
+		}
+
+		// Dynamic Value
+		if strings.HasPrefix(capacity, "@") {
+			// Remove the @ prefix
+			capacity = strings.TrimPrefix(capacity, "@")
+			// capacity is a string in the form of attribute.featureset.elements
+			split := strings.SplitN(capacity, ".", 3)
+			if value, ok := features.Attributes[split[0]+"."+split[1]]; ok {
+				q, err := k8sQuantity.ParseQuantity(value.Elements[split[2]])
+				if err != nil {
+					klog.Errorf("bad label value %s encountered for extended resource: %s", q.String(), extendedResource, err.Error())
+					continue
+				}
+
+				extendedResources[extendedResource] = q.String()
+				continue
+			}
+			continue
+		}
+		// Static Value (Pre-Defined at the NodeFeatureRule)
+		q, err := k8sQuantity.ParseQuantity(capacity)
+		if err != nil {
+			klog.Errorf("bad label value %s encountered for extended resource: %s", capacity, extendedResource, err.Error())
+			continue
+		}
+		extendedResources[extendedResource] = q.String()
+		continue
```

Comment:
Maybe we wouldn't need so many `continue`s if we put this inside else
```suggestion
		} else {
			// Static Value (Pre-Defined at the NodeFeatureRule)
			q, err := k8sQuantity.ParseQuantity(capacity)
			if err != nil {
				klog.Errorf("bad label value %s encountered for extended resource: %s", capacity, extendedResource, err.Error())
				continue
		}
		extendedResources[extendedResource] = q.String()
```